content/en/docs/reference/access-authn-authz/extensible-admission-controllers.md

```diff
@@ -1300,6 +1300,106 @@ Admission webhooks created using `admissionregistration.k8s.io/v1beta1` default
+   suffixed with the current timestamp (e.g. `foo-sidecar-19700101-000000`).
+
+2. For a `CREATE`/`UPDATE` pod request, reject if the pod has label `"env"` set,
+   otherwise add an `"env": "prod"` label to the pod.
+
+3. For a `CREATE` pod request, blindly append a sidecar container named
+   `foo-sidecar` without looking to see if there is already a `foo-sidecar`
+   container in the pod.
+
+In the first case above, reinvoking the webhook can result in the same sidecar being injected multiple times to a pod, each time
+with a different container name. Similarly the webhook can inject duplicated containers if the sidecar already exists in
+a user-provided pod.
+
+In the second case above, reinvoking the webhook will result in the webhook failing on its own output.
+
+In the third case above, reinvoking the webhook will result in duplicated containers in the pod spec, which makes
+the request invalid and rejected by the API server.
+
+### Intercepting all versions of an object
+
+It is recommended that admission webhooks should always intercept all versions of an object by setting `.webhooks[].matchPolicy`
+to `Equivalent`. It is also recommended that admission webhooks should prefer registering for stable versions of resources.
+Failure to intercept all versions of an object can result in admission policies not being enforced for requests in certain
+versions. See [Matching requests: matchPolicy](#matching-requests-matchpolicy) for examples.
+
+### Availability
+
+It is recommended that admission webhooks should evaluate as quickly as possible (typically in milliseconds), since they add to API request latency.
+It is encouraged to use a small timeout for webhooks. See [Timeouts](#timeouts) for more detail.
+
+It is recommended that admission webhooks should leverage some format of load-balancing, to provide high availability and
+performance benefits. If a webhook is running within the cluster, you can run multiple webhook backends behind a service
+to leverage the load-balancing that service supports.
+
+### Guaranteeing the final state of the object is seen
+
+Admission webhooks that need to guarantee they see the final state of the object in order to enforce policy
+should use a validating admission webhook, since objects can be modified after being seen by mutating webhooks.
+
+For example, a mutating admission webhook is configured to inject a sidecar container with name "foo-sidecar" on every
+`CREATE` pod request. If the sidecar *must* be present, a validating admisson webhook should also be configured to intercept `CREATE` pod requests, and validate
+that a container with name "foo-sidecar" with the expected configuration exists in the to-be-created object.
+
+### Avoiding deadlocks in self-hosted webhooks
+
+A webhook that enacts policies for webhook mission-critical resources might cause deadlocks for its own deployment if it is running inside the cluster.
```

Comment:
"webhook mission-critical resources" is hard to understand. suggest rephrasing to:

> A webhook running inside the cluster might cause deadlocks for its own deployment if it is configured to intercept resources required to start its own pods.